delphin/eds/_operations.py

```diff
@@ -64,12 +65,17 @@ def from_mrs(m, predicate_modifiers=False, unique_ids=True,
     return e
 
 
-def _mrs_get_top(top, hcmap, reps):
+def _mrs_get_top(top, hcmap, reps, index, ivmap):
     if top in hcmap:
         lbl = hcmap[top].lo
-        top = reps[lbl][0].id
+        if lbl in reps:
+            top = reps[lbl][0].id
+        else:
+            top = reps[ivmap[index].label][0].id
```

Comment:
I think the logic is wrong here because of how it's nested and how it gets the EP for an INDEX. First, it checks the INDEX iff there is a TOP that is used in a qeq. Maybe something like this would be better:

```python
if top in hcmap and hcmap[top].lo in reps:
    lbl = hcmap[top].lo
    top = reps[lbl][0].id
elif top in reps:
    top = reps[top][0].id
elif index in ivmap:
    top = ivmap[index]  #  but see my comment about creating ivmap
else:
    top = None
```

In the LKB's `dependencies.lisp` it looks like the code to find a representative does something special for non-handle variables, but it doesn't look like it uses the label. I think it collects all EPs that aren't quantifiers or "bleached" and share the intrinsic variable, then selects among those. Except for the bleached predicates this is doable an a grammar-agnostic way, although I don't like it much because having multiple non-quantifier EPs sharing an intrinsic variable is ill-formed (some might disagree).